lib/editor/util/validation.js

```diff
@@ -36,22 +36,43 @@ export function validate (
   const valueDoesNotExist = doesNotExist(value)
   const isRequiredButEmpty = required && valueDoesNotExist
   const isOptionalAndEmpty = !required && valueDoesNotExist
-  let reason = 'Required field must not be empty'
   const agencies = getTableById(tableData, 'agency')
+  let locationType: ?number = null
+
+  // entity.locationtype is a string.  Convert to number for conditinals later on.
+  if (entity && entity.location_type !== null) {
+    locationType = parseInt(entity.location_type, 10)
+  }
 
   // setting as a variable here because of eslint bug
   type CheckPositiveOutput = {
     num?: number,
     result: false | EditorValidationIssue
   }
 
+  /**
+   * Construct an EditorValidationIssue for the field name and reason (defaults to
+   * empty field message).
+   */
+  function validationIssue (reason: string, field = name) {
+    return {field, invalid: true, reason}
+  }
+
+  /**
+   * Construct an EditorValidationIssue for this field, used if it is required
+   * and has an empty value.
+   */
+  function emptyFieldValidationIssue (field = name) {
+    return validationIssue('Required field must not be empty', field)
+  }
+
```

Comment:
Please move these two functions to the root level of the class since they don't require on variables scoped inside the `validate` function.